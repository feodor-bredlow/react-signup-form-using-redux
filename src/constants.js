export const USER_INFO = {
	FIRST_NAME: 'firstName',
	LAST_NAME: 'lastName',
	NICK_NAME: 'nickName',
	EMAIL: 'email',
	PASSWORD: 'password',
	REPEAT_PASSWORD: 'repeatPassword',
	SHOW_ADDRESS: 'showAddress',
	STREET: 'street',
	HOUSE_NR: 'houseNr',
	ZIP: 'zip',
	CITY: 'city',
	ADDITIONAL_INFO: 'additionalInfo',
};
