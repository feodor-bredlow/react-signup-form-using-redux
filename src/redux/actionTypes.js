export const UPDATE_SHOW_ADDRESS = 'UPDATE_SHOW_ADDRESS';
