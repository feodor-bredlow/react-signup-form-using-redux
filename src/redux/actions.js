import {
	UPDATE_FIRST_NAME,
	UPDATE_LAST_NAME,
	UPDATE_NICK_NAME,
	UPDATE_EMAIL,
	UPDATE_PASSWORD,
	UPDATE_REPEAT_PASSWORD,
	UPDATE_SHOW_ADDRESS,
	UPDATE_STREET,
	UPDATE_HOUSENR,
	UPDATE_ZIP,
	UPDATE_CITY,
	UPDATE_ADDITIONAL_INFO,
} from './actionTypes';

// user
export const updateFirstName = (firstName) => ({
	type: UPDATE_FIRST_NAME,
	payload: {
		firstName,
	},
});
export const updateLastName = (lastName) => ({
	type: UPDATE_LAST_NAME,
	payload: {
		lastName,
	},
});
export const updateNickName = (nickName) => ({
	type: UPDATE_NICK_NAME,
	payload: {
		nickName,
	},
});
export const updateEmail = (email) => ({
	type: UPDATE_EMAIL,
	payload: {
		email,
	},
});
export const updatePassword = (password) => ({
	type: UPDATE_PASSWORD,
	payload: {
		password,
	},
});
export const updateRepeatPassword = (repeatPassword) => ({
	type: UPDATE_REPEAT_PASSWORD,
	payload: {
		repeatPassword,
	},
});

// Address actions
export const updateShowAddress = (showAddress) => ({
	type: UPDATE_SHOW_ADDRESS,
	payload: {
		showAddress,
	},
});
export const updateStreet = (street) => ({
	type: UPDATE_STREET,
	payload: {
		street,
	},
});
export const updateHouseNr = (houseNr) => ({
	type: UPDATE_HOUSENR,
	payload: {
		houseNr,
	},
});
export const updateZip = (zip) => ({
	type: UPDATE_ZIP,
	payload: {
		zip,
	},
});
export const updateCity = (city) => ({
	type: UPDATE_CITY,
	payload: {
		city,
	},
});

// Additional information actions
export const updateAdditionalInfo = (additionalInfo) => ({
	type: UPDATE_ADDITIONAL_INFO,
	payload: {
		additionalInfo,
	},
});
