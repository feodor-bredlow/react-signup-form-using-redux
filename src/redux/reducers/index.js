import { combineReducers } from 'redux';
import userInfo from './userInfo';
import address from './address';

export default combineReducers({ userInfo, address });
