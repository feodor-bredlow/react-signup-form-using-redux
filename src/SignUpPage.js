import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import * as actions from './redux/actions';
import * as regex from './utils/regex';

import { USER_INFO } from './constants';
import styles from './SignUpPage.module.scss';
import DefaultForm from './components/DefaultForm/DefaultForm.js';
import DefaultInput from './components/DefaultInput/DefaultInput.js';
import CheckboxInput from './components/CheckboxInput/CheckboxInput.js';

const SignUpPage = (props) => {
	return (
		<div className={styles.wrapper}>
			<DefaultForm>
				<DefaultInput
					label="Last name"
					identifier={USER_INFO.LAST_NAME}
				/>
				<DefaultInput
					label="First name"
					identifier={USER_INFO.FIRST_NAME}
				/>
				<DefaultInput
					label="Nick name"
					identifier={USER_INFO.NICK_NAME}
				/>
				<DefaultInput
					label="Email"
					identifier={USER_INFO.EMAIL}
					type="email"
				/>
				<DefaultInput
					label="Password"
					identifier={USER_INFO.PASSWORD}
					type="password"
				/>
				<DefaultInput
					label="Repeat Password"
					identifier={USER_INFO.REPEAT_PASSWORD}
					type="password"
				/>
				<CheckboxInput
					label="Show address"
					identifier={USER_INFO.SHOW_ADDRESS}
				/>
				{props.store.address.showAddress && (
					<>
						<DefaultInput
							label="Street"
							identifier={USER_INFO.STREET}
						/>
						<DefaultInput
							label="House number"
							identifier={USER_INFO.HOUSE_NR}
							notRequired
						/>
						<DefaultInput
							label="ZIP"
							identifier={USER_INFO.ZIP}
							maxLength={5}
							regex={regex.digit()}
						/>
						<DefaultInput
							label="City"
							identifier={USER_INFO.CITY}
						/>
					</>
				)}
				<DefaultInput
					label="Additional information"
					multiLine
					identifier={USER_INFO.ADDITIONAL_INFO}
				/>
			</DefaultForm>
		</div>
	);
};

SignUpPage.propTypes = {
	store: PropTypes.object,
	actions: PropTypes.object,
};

const mapStateToProps = (state) => {
	return { store: state };
};

const mapDispatchToProps = (dispatch) => ({
	actions: bindActionCreators(actions, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(SignUpPage);
