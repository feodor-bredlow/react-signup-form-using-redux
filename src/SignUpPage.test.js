import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render, cleanup } from '@testing-library/react';
import SignUpPage from './SignUpPage';
import combinedReducers from './redux/reducers/index';
import { USER_INFO } from './constants';

describe('SignUpForm', () => {
	let query;

	afterEach(cleanup);

	describe('initial page load with empty values', () => {
		let userInfo = {
			[USER_INFO.LAST_NAME]: '',
			[USER_INFO.FIRST_NAME]: '',
			[USER_INFO.NICK_NAME]: '',
			[USER_INFO.EMAIL]: '',
			[USER_INFO.PASSWORD]: '',
			[USER_INFO.REPEAT_PASSWORD]: '',
			[USER_INFO.STREET]: '',
			[USER_INFO.HOUSE_NR]: '',
			[USER_INFO.ZIP]: '',
			[USER_INFO.CITY]: '',
			[USER_INFO.ADDITIONAL_INFO]: '',
		};

		let address = {
			showAddress: true,
		};

		let initialState = { userInfo, address };

		let store = createStore(combinedReducers, initialState);

		beforeEach(() => {
			query = render(
				<Provider store={store}>
					<SignUpPage />
				</Provider>
			);
		});

		it('SignUpForm should render', () => {
			expect(query.getByTestId('signup-form')).not.toBeNull();
			expect(query.getByTestId('signup-form')).toMatchSnapshot();
		});

		it('submit button should initially be disabled', () => {
			expect(query.getByTestId('button-submit')).toBeDisabled();
		});
	});

	describe('page behaviour when input is filled', () => {
		let userInfo = {
			[USER_INFO.LAST_NAME]: 'nachname',
			[USER_INFO.FIRST_NAME]: 'vorname',
			[USER_INFO.NICK_NAME]: 'a',
			[USER_INFO.EMAIL]: 'asfd@asfd.com',
			[USER_INFO.PASSWORD]: '123asd',
			[USER_INFO.REPEAT_PASSWORD]: '123asd',
			[USER_INFO.STREET]: 'strasse',
			[USER_INFO.HOUSE_NR]: '123a',
			[USER_INFO.ZIP]: '12345',
			[USER_INFO.CITY]: 'stadt',
			[USER_INFO.ADDITIONAL_INFO]: 'more information',
		};

		let address = {
			showAddress: true,
		};

		let initialState = { userInfo, address };

		let store = createStore(combinedReducers, initialState);
		beforeEach(() => {
			query = render(
				<Provider store={store}>
					<SignUpPage />
				</Provider>
			);
		});

		it('submit button should be enabled', () => {
			expect(query.getByTestId('button-submit')).toBeEnabled();
		});
	});
});
