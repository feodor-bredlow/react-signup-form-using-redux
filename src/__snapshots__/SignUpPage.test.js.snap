// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm SignUpForm should render 1`] = `
<div
  class="wrapper"
  data-testid="signup-form"
>
  <form>
    <div>
      <div>
        <div>
          <div
            class="inputOuterWrapper"
          >
            <label
              class="formLabel"
            >
              Last name
               
              <span>
                *
              </span>
            </label>
            <input
              class="inputElem"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div>
          <div
            class="inputOuterWrapper"
          >
            <label
              class="formLabel"
            >
              First name
               
              <span>
                *
              </span>
            </label>
            <input
              class="inputElem"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div>
          <div
            class="inputOuterWrapper"
          >
            <label
              class="formLabel"
            >
              Nick name
               
              <span>
                *
              </span>
            </label>
            <input
              class="inputElem"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div>
          <div
            class="inputOuterWrapper"
          >
            <label
              class="formLabel"
            >
              Email
               
              <span>
                *
              </span>
            </label>
            <input
              class="inputElem"
              type="email"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div>
          <div
            class="inputOuterWrapper"
          >
            <label
              class="formLabel"
            >
              Password
               
              <span>
                *
              </span>
            </label>
            <input
              class="inputElem"
              type="password"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div>
          <div
            class="inputOuterWrapper"
          >
            <label
              class="formLabel"
            >
              Repeat Password
               
              <span>
                *
              </span>
            </label>
            <input
              class="inputElem"
              type="password"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div
          class="checkboxWrapper"
        >
          <input
            checked=""
            type="checkbox"
          />
          Show address
        </div>
      </div>
      <div>
        <div>
          <div
            class="inputOuterWrapper"
          >
            <label
              class="formLabel"
            >
              Street
               
              <span>
                *
              </span>
            </label>
            <input
              class="inputElem"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div>
          <div
            class="inputOuterWrapper"
          >
            <label
              class="formLabel"
            >
              House number
               
            </label>
            <input
              class="inputElem"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div>
          <div
            class="inputOuterWrapper"
          >
            <label
              class="formLabel"
            >
              ZIP
               
              <span>
                *
              </span>
            </label>
            <input
              class="inputElem"
              maxlength="5"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div>
          <div
            class="inputOuterWrapper"
          >
            <label
              class="formLabel"
            >
              City
               
              <span>
                *
              </span>
            </label>
            <input
              class="inputElem"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div>
          <div
            class="inputOuterWrapper"
          >
            <label
              class="formLabel"
            >
              Additional information
               
              <span>
                *
              </span>
            </label>
            <textarea
              class="inputElem"
              name="Text1"
              rows="5"
            />
          </div>
        </div>
      </div>
    </div>
    <input
      class="submitButton"
      data-testid="button-submit"
      disabled=""
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;

exports[`SignUpForm initial page load with empty values SignUpForm should render 1`] = `
<div
  class="wrapper"
  data-testid="signup-form"
>
  <form>
    <div>
      <div>
        <div>
          <div
            class="inputOuterWrapper"
          >
            <label
              class="formLabel"
            >
              Last name
               
              <span>
                *
              </span>
            </label>
            <input
              class="inputElem"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div>
          <div
            class="inputOuterWrapper"
          >
            <label
              class="formLabel"
            >
              First name
               
              <span>
                *
              </span>
            </label>
            <input
              class="inputElem"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div>
          <div
            class="inputOuterWrapper"
          >
            <label
              class="formLabel"
            >
              Nick name
               
              <span>
                *
              </span>
            </label>
            <input
              class="inputElem"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div>
          <div
            class="inputOuterWrapper"
          >
            <label
              class="formLabel"
            >
              Email
               
              <span>
                *
              </span>
            </label>
            <input
              class="inputElem"
              type="email"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div>
          <div
            class="inputOuterWrapper"
          >
            <label
              class="formLabel"
            >
              Password
               
              <span>
                *
              </span>
            </label>
            <input
              class="inputElem"
              type="password"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div>
          <div
            class="inputOuterWrapper"
          >
            <label
              class="formLabel"
            >
              Repeat Password
               
              <span>
                *
              </span>
            </label>
            <input
              class="inputElem"
              type="password"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div
          class="checkboxWrapper"
        >
          <input
            checked=""
            type="checkbox"
          />
          Show address
        </div>
      </div>
      <div>
        <div>
          <div
            class="inputOuterWrapper"
          >
            <label
              class="formLabel"
            >
              Street
               
              <span>
                *
              </span>
            </label>
            <input
              class="inputElem"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div>
          <div
            class="inputOuterWrapper"
          >
            <label
              class="formLabel"
            >
              House number
               
            </label>
            <input
              class="inputElem"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div>
          <div
            class="inputOuterWrapper"
          >
            <label
              class="formLabel"
            >
              ZIP
               
              <span>
                *
              </span>
            </label>
            <input
              class="inputElem"
              maxlength="5"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div>
          <div
            class="inputOuterWrapper"
          >
            <label
              class="formLabel"
            >
              City
               
              <span>
                *
              </span>
            </label>
            <input
              class="inputElem"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div>
          <div
            class="inputOuterWrapper"
          >
            <label
              class="formLabel"
            >
              Additional information
               
              <span>
                *
              </span>
            </label>
            <textarea
              class="inputElem"
              name="Text1"
              rows="5"
            />
          </div>
        </div>
      </div>
    </div>
    <input
      class="submitButton"
      data-testid="button-submit"
      disabled=""
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;
